File: src/main/java/com/kelsos/mbrc/model/UserAction.java
package com.kelsos.mbrc.model;

public class UserAction {
    private String context;
    private Object data;

    public UserAction(String context, Object data) {
        this.context = context;
        this.data = data;
    }

    public String getContext() {
        return context;
    }

    public Object getData() {
        return data;
    }
}

File: core/networking/src/main/kotlin/com/kelsos/mbrc/core/networking/protocol/actions/UserAction.kt
package com.kelsos.mbrc.core.networking.protocol.actions

import com.kelsos.mbrc.core.networking.protocol.base.Protocol

data class UserAction(val protocol: Protocol, val data: Any) {
  companion object {
    fun create(protocol: Protocol): UserAction = UserAction(protocol, true)

    fun create(protocol: Protocol, data: Any): UserAction = UserAction(protocol, data)

    fun toggle(protocol: Protocol): UserAction = UserAction(protocol, Protocol.TOGGLE)
  }
}
